### DISCUSSION TITLE
Spark on Nebula Graph - siwei.io

### DISCUSSION THREAD

**Discussion by giscus** [siwei-io/siwei-io.github.io#7] [Comments]
# Spark on Nebula Graph - siwei.io

What could be done with Spark and PySpark on top of Nebula Graph, this post covers everything we should know.

https://www.siwei.io/en/spark-on-nebula-graph/

**Comment by wey-gu** [comment]
If you would like to use another version of spark env, check their tags here: https://hub.docker.com/r/bde2020/spark-master/tags?page=1

and replace the image tag accordingly.

**Comment by manyfeatures** [comment]
Great post!

I had to replace the spark connector docker network to `nebula-net` in the code below in order to connect to Nebula from pyspark.
```bash
docker run --name spark-master-0 --network nebula-docker-compose_nebula-net \
    -h spark-master-0 -e ENABLE_INIT_DAEMON=false -d \
    -v ${PWD}/:/root \
    bde2020/spark-master:2.4.5-hadoop2.7
```
Because it seems to be the default nebula graph network according to docker compose file

**Reply by wey-gu** [reply]
Thanks @manyfeatures :), indeed, the docker network created with docker-compose would vary in different version of docker-compose/docker version/distribution.

Thanks for the sharing :).
